### DISCUSSION TITLE
Can Olivia use database for dynamically adding Intents

### DISCUSSION THREAD

**Discussion by lonnietc** [olivia-ai/olivia#162] [Ideas]
Hello,

I'm a bit new to Olivia and am trying to see what the capabilities are in terms of dynamically adding more Intents while operational which led me to the question as to if Olivia can use a database (SQL, Graph, or other) to have Intents dynamically added while online?

I am guessing that this is not the case because Olivia might need re-training after each updated to include more Intents, etc. but wanted to check on this just in case.

Thanks

**Comment by hugolgst** [comment]
Hi,

Indeed we do not have any kind of database, since we would, in any case, need a re-training of the model.
We can however add this to the backlog of the v3 that I am currently starting.

The only potential issue I see with this is that we could have a lot of data in the database and might be annoying for existing .csv to dump the content to a database.
